daemon/inspect.go

```diff
@@ -166,11 +166,17 @@ func (daemon *Daemon) getInspectData(container *container.Container, size bool)
 
 	contJSONBase.GraphDriver.Name = container.Driver
 
-	graphDriverData, err := container.RWLayer.Metadata()
-	if err != nil {
-		return nil, err
+	// If container is marked as Dead, the container's graphdriver metadata
+	// could have been removed, it will case error if we try to get the metadata.
+	if container.Dead {
+		contJSONBase.GraphDriver.Data = nil
+	} else {
+		graphDriverData, err := container.RWLayer.Metadata()
```

Comment:
How about instead of the `if ;else` we always try to grab the driver data, but `if err != nil && !container.Dead { return err }`